compiler/src/dmd/dtemplate.d

```diff
@@ -6293,6 +6300,8 @@ extern (C++) class TemplateInstance : ScopeDsymbol
                     minst = tnext.minst; // cache result from non-speculative sibling
                     return false;
                 }
+                else if (needsCodegen != ThreeState.none)
+                    break;
```

Comment:
Codecov says this is not covered, is it Codecov's issue or really not covered ?

Reply:
It's really not covered, because the interior function never returns `ThreeState.no`.

I left it in to guard us in case that were to change in the future, and also to align with the non-allinst branch.

I _think_ these two branches could be factored because the outer shell is mostly a duplicate now, but I'll leave that for something to look at in far future - maybe one of the strategies will have to go.